src/arith/analyzer.cc

```diff
@@ -108,29 +108,25 @@ bool Analyzer::CanProveEqual(const PrimExpr& lhs, const PrimExpr& rhs) {
 }
 
 bool Analyzer::CanProve(const PrimExpr& expr) {
-  if (const auto* ptr = expr.as<IntImmNode>()) {
-    return ptr->value != 0;
```

Comment:
These are quick path that we intended leave as it is to avoid trigerring of recursive function calling cost. Given CanProve can be called frequently, it would be useful to keep it as it is in this case.

We can document the rationle using comments

Reply:
Sounds good, and added back in.  I hadn't done any performance tests, but had been expecting that the call to Simplify would be low cost as it starts by checking whether the value is a constant.

Reply:
I agree with what you said(simplify also starts to be low cost), so I could be over simplifying here :) In the meantime being explicit without relying on subsequent behavior (which would be one cost of call) can be helpful a bit.

Reply:
hmm i still see the two lines deleted, perhaps a miss that forgets to checkin?